### PR TITLE
Add custom boolean attribute support #503

### PR DESCRIPTION
**Pull request by Legioth**
* Adds BooleanAttribute that writes out itself without a value
* Adds API in Element for setting boolean attributes
* Update parser to distinguish between no value and empty value

### CODE REVIEW

File: src/test/java/org/jsoup/nodes/ElementTest.java [hunk 1]
```diff
@@ -294,6 +294,22 @@
...
+        div.attr("false", "value");
+        div.attr("false", false);
+        
+        assertTrue(div.hasAttr("true"));
+        assertEquals("", div.attr("true"));
+        
+        assertFalse(div.hasAttr("false"));
+ 
+        assertEquals("<div true></div>", div.outerHtml());
+    }
```
Comment (jhy) [inline src/test/java/org/jsoup/nodes/ElementTest.java]: Can you assert their type here also?

File: src/test/java/org/jsoup/parser/AttributeParseTest.java [hunk 2]
```diff
@@ -66,4 +66,16 @@
         Elements els = Jsoup.parse(html).select("a");
         assertEquals("&wr_id=123&mid-size=true&ok=&wr", els.first().attr("href"));
     }
+    
+    @Test public void parsesBooleanAttributes() {
```
Comment (jhy) [inline src/test/java/org/jsoup/parser/AttributeParseTest.java]: Assert type here also

**Comment by jhy** [comment]
Thanks, looks interesting. Can you check what the behavior should be for XML parsing and XML syntax output? Thinking that at least for XML syntax output, boolean attributes should actually output as empty string attributes.

**Comment by Legioth** [comment]
From what I have found, the original XHTML specification only allowed the format `attribute="attribute"`.

Latest XHTML specs also allow `attribute=""` according to e.g. https://html.spec.whatwg.org/#boolean-attributes

For attributes defined as boolean using XML Schema, the format is `attribute="true"` or `attribute="1"`  (not compatible with XHTML version).

**Comment by Legioth** [comment]
Carefully bumping this.

**Comment by jhy** [comment]
Thanks! Merged.

**Pull request merged**